### ISSUE TITLE
iOS outline app is not accessible for the blind!

### ISSUE THREAD

**Issue by denizsincar29** [Jigsaw-Code/outline-client#2670]
### Application

Outline Client

### Describe the bug

Hello! I am a voiceover user. Voiceover is a screenreader for blind that reads things on the screen and makes me able to use my iPhone without sight. The problem I am facing with outline client is some buttons are announced just as "button", not "menu button", "settings button", and another problem is that i cannot press on any of these buttons using voiceover's activate feature. So i focus a button, double tap it to activate and it doesn't do anything. It means not native iOS GUI is used, but some workarounded selfmade gui.
My sighted friends even told me that where the buttons are recognized using voiceover and where the buttons actually are is totally different. E.g. when i explore left lower part of screen center, the entire part is announced as connect button which connects to the vpn. But my friends told me that it's at far right part, where voiceover makes "empty" sound "dit dit dit".
Please take my issue seriously and give me advise how to turn off the vpn now from control center?

### Steps to reproduce

1. Turn on voiceover from accessibility settings.
2. Open Outline application.
3. Try to do anything in the app. You will get misplaced buttons.

### What did you expect to happen?

All buttons must be labled and working correctly.

### What actually happened?

Misplaced buttons, sometimes with no label or accessibility hint.

### Outline Version

all versions from september to 20 november 2025.

### What operation system are you using?

iOS

### Operating System Version

IOS 26.2 I guess.

**Comment by jyyi1** [comment]
Hi @denizsincar29 ,

Thanks for raising this issue. We're really sorry about the accessibility problems and we'll definitely look into it.

In the meantime, we have a workaround that should help. Once you've connected to a key successfully for the first time within the app, you can use the iOS system VPN UI to connect and disconnect from the VPN going forward. You can do this from the `Settings` > `General` > `VPN & Device Management` (or by searching for VPN), which should be fully accessible.

This should allow you to use Outline without having to navigate the in-app buttons that are causing trouble.

We'll continue to work on a proper fix for the app itself and will post updates here. Thanks for your patience and for helping us improve Outline.
